### PR TITLE
Use GoReleaser for building and publishing the images [ONPREM-1263]

### PR DESCRIPTION
**Pull request by christian-stephen**
:gear: **Issue**

**Ticket/s**:

---

:gear: **Change Description**

**Change Type**:

*Why this change? Reference the ticket and acceptance criteria if any. Specify the type of change: bug fix, new feature, breaking change, documentation update, security, etc.*

**Acceptance Criteria**:

---

:white_check_mark: **Solution**

*What was the solution? How did you fix the issue or implement the new feature?*

---

:question: **Testing**

*Describe what was tested. Remember to include changes in functionality, edge cases, and enough detail that another developer can replicate your progress.*

- [ ] Created and updated tests where applicable

---

:book: **Documentation Updates**

*Have any updates been made to the documentation?*

- [ ] Updated related documentation, if applicable
- [ ] Updated [changelog](https://github.com/circleci/runner-init/blob/main/CHANGELOG.md)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,47 +1,26 @@
 version: 2.1
 
-parameters:
-  release-name:
-    type: string
-    default: "runner-init"
```
Comment (christian-stephen) [inline .circleci/config.yml:6]: I think we decided not to use an different repo for dev builds? The codebase is public, so no point for this parameter really. We can add this back if needed in the future.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,47 +1,26 @@
...
   main-workflow:
     jobs:
       - lint
       - build
-      - build-and-publish-images:
-          name: build-and-publish-image-amd64
-          context: org-global
+      - images:
+          context: [ org-global ]
           requires: [ lint, build ]
```
Comment (christian-stephen) [inline .circleci/config.yml:23]: These images are really small, and we use xbuild to build multi-arch images, so let's just do this all within a single job/machine.

File: docker/Dockerfile
Comment (christian-stephen) [inline docker/Dockerfile:1]: In a followup PR, I'll add a mode to the orchestrator to do the copying and linking instead of a shell script. This enables us to use a `scratch` image.

File: .circleci/config.yml (hunk 1 above)
Comment (christian-stephen) [reply .circleci/config.yml:6]: Also, we might want to publish dev tags in the future too. If we want to incorporate them in smoke tests and such.

File: .circleci/config.yml [hunk 3]
```diff
@@ -93,39 +72,16 @@ jobs:
...
     steps:
-      - checkout
+      - goreleaser_setup
       - docker_login
-      - run: ./do publish-docker-manifest << pipeline.parameters.release-name >>
-      - notify_failing_main
+      - run:
+          name: Build and maybe push images
+          command: |
+            SKIP_PUSH=$(if [ "$CIRCLE_BRANCH" == "main" ]; then echo "false"; else echo "true"; fi) \
```
Comment (christian-stephen) [inline .circleci/config.yml:83]: Images are only pushed on `main`.
